app/src/main/java/com/github/h3lp3rs/h3lp/messaging/RecentMessagesActivity.kt

```diff
@@ -0,0 +1,133 @@
+package com.github.h3lp3rs.h3lp.messaging
+
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.github.h3lp3rs.h3lp.EXTRA_HELPEE_ID
+import com.github.h3lp3rs.h3lp.EXTRA_USER_ROLE
+import com.github.h3lp3rs.h3lp.R
+import com.github.h3lp3rs.h3lp.database.Databases
+import com.github.h3lp3rs.h3lp.database.Databases.Companion.databaseOf
+import com.github.h3lp3rs.h3lp.messaging.Messenger.HELPEE
+import com.xwray.groupie.GroupAdapter
+import com.xwray.groupie.Item
+import com.xwray.groupie.ViewHolder
+import kotlinx.android.synthetic.main.activity_latest_messages.*
+import kotlinx.android.synthetic.main.latest_message_row.view.*
+import java.util.*
+
+
+class RecentMessagesActivity : AppCompatActivity() {
+    private var helpeeId : String? = null
+    private val adapter = GroupAdapter<ViewHolder>()
+
+    // Database where all the conversation ids generated by helpers will be stored
+    private val conversationIdsDb = databaseOf(Databases.CONVERSATION_IDS)
+    private val messagesDatabase = databaseOf(Databases.MESSAGES)
+
+    /**
+     * The map is a synchronizedMap to allow for several conversations to be concurrently added
+     * to it
+     */
+    private var idToItem = Collections.synchronizedMap<String, HelperConversation>(mutableMapOf())
+
+    override fun onCreate(savedInstanceState : Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_latest_messages)
+
+        val bundle = this.intent.extras
+        helpeeId = bundle?.getString(EXTRA_HELPEE_ID) ?: helpeeId
+
+        databaseOf(Databases.MESSAGES).setString("HEWIEWEWE", helpeeId!!)
```

Comment:
You can directly use the messagesDatabase attribute (Totally understood the message string 😜)

Reply:
Haha agree, but that wasn't done by me, it should be corrected in #201